### PR TITLE
[#IP-308] Fix Subscription Feed on settings mode change

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: UpdateSubscriptionsFeedActivity/handler.ts [hunk 1]
```diff
@@ -92,21 +109,58 @@ export const updateSubscriptionFeed = async (
...
+              rowKey: `${uPreferencePartitionKey}-${fiscalCodeHash}`
+            }
+          ];
+        },
+        [] as ReadonlyArray<SubscriptionFeedEntitySelector>
+      )
+    )
+    .getOrElse([]);
+
+  const doenstInsertIfDeleted = decodedInput.subscriptionKind === "SERVICE";
```
Comment (AleDore) [inline UpdateSubscriptionsFeedActivity/handler.ts:140]: typo :) 
```suggestion
  const doesntInsertIfDeleted = decodedInput.subscriptionKind === "SERVICE";
```

Comment (AleDore) [inline UpdateSubscriptionsFeedActivity/handler.ts:140]: Anyway I'd rather name this with a positive meaning. Something like `allowInsertIfDeleted`.

File: utils/subscription_feed.ts [hunk 2]
```diff
@@ -34,41 +45,74 @@ export const updateSubscriptionStatus = (
...
   logPrefix: string,
   version: number,
-  delPartitionKey: string,
-  delKey: string,
-  insPartitionKey: string,
-  insKey: string
+  deleteEntity: SubscriptionFeedEntitySelector,
+  deleteOtherEntities: ReadonlyArray<SubscriptionFeedEntitySelector>,
+  insertEntity: SubscriptionFeedEntitySelector,
+  doesntInsertIfDeleted: boolean
```
Comment (AleDore) [inline utils/subscription_feed.ts:51]: ditto

File: UpsertedProfileOrchestrator/__tests__/handler.test.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ const someRetryOptions = new df.RetryOptions(5000, 10);
 someRetryOptions.backoffCoefficient = 1.5;
 
 // tslint:disable-next-line: no-big-function
-describe("UpsertedProfileOrchestrator", () => {
+describe("UpsertedProfileOrchestratorV2", () => {
```
Comment (AleDore) [inline UpsertedProfileOrchestrator/__tests__/handler.test.ts:33]: Why?

Comment (gquadrati) [reply UpsertedProfileOrchestrator/__tests__/handler.test.ts:33]: Ouch, a wrong replace.. thanks

Comment (gquadrati) [reply UpsertedProfileOrchestrator/__tests__/handler.test.ts]: Fixed in [0a20b29](https://github.com/pagopa/io-functions-app/pull/189/commits/0a20b298929d320f91a74b0302eb9aefb54ccf7e)

File: utils/subscription_feed.ts (hunk 2 above)
Comment (gquadrati) [reply utils/subscription_feed.ts]: Fixed in [0a20b29](https://github.com/pagopa/io-functions-app/pull/189/commits/0a20b298929d320f91a74b0302eb9aefb54ccf7e)

File: UpdateSubscriptionsFeedActivity/handler.ts (hunk 1 above)
Comment (gquadrati) [reply UpdateSubscriptionsFeedActivity/handler.ts]: Fixed in [0a20b29](https://github.com/pagopa/io-functions-app/pull/189/commits/0a20b298929d320f91a74b0302eb9aefb54ccf7e)

**Comment by AleDore** [comment]
@gquadrati @fabriziopapi : Why Draft?

**Comment by gquadrati** [comment]
> @gquadrati @fabriziopapi : Why Draft?

Actually because we didn't realize it was a Draft :)

**Review by AleDore — APPROVED** [review]
LGTM

**Review by fabriziopapi — APPROVED** [review]

**Pull request merged**